core/src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -69,6 +68,15 @@ public void clear() {
         return Collections.unmodifiableSet(this.connections);
     }
 
+    /*
+     * @return The current per-project settings.  This object may become out of date if the settings are edited
+     * by the user, so objects requiring a preference value should also subscribe to {@link ProjectSettingsChangedEvent}
+     * to get updates.
+     */
+    public ProjectSettings getProjectSettings() {
+        return settings;
```

Comment:
You get the value of ensuring that updates to the object can't happen without a `ProjectStettingsChangedEvent` from being posted.